File: abc/083/b/cpp/main.cpp
#include <iostream>

using namespace std;

int sumOfDigits(int x)
{
    int sum = 0;
    while (x > 0)
    {
        sum += x % 10;
        x /= 10;
    }
    return sum;
}

int main()
{
    int rangeMax, digitSumMin, digitSumMax;
    cin >> rangeMax >> digitSumMin >> digitSumMax;

    int sum = 0;
    for (int i = 1; i <= rangeMax; ++i)
    {
        int digitSum = sumOfDigits(i);
        if (digitSumMin <= digitSum && digitSum <= digitSumMax)
        {
            sum += i;
        }
    }

    cout << sum << endl;
}

File: abc/083/b/go/main.go
package main

import (
	"fmt"
)

func digitSum(n int) int {
	sum := 0
	for n > 0 {
		sum += n % 10
		n /= 10
	}
	return sum
}

func main() {
	var rangeMax, digitSumMin, digitSumMax int
	fmt.Scan(&rangeMax, &digitSumMin, &digitSumMax)

	sum := 0
	for i := 1; i <= rangeMax; i++ {
		dsum := digitSum(i)
		if digitSumMin <= dsum && dsum <= digitSumMax {
			sum += i
		}
	}

	fmt.Println(sum)
}
